File: docs/code-examples/all/scalar_simple.py
"""Log a scalar over time."""
import math

import rerun as rr

rr.init("rerun_example_scalar", spawn=True)

# Set up plot styling: Logged timeless since it never changes and affects all timelines.
rr.log("scalar", rr.SeriesPoint(), timeless=True)

# Log the data on a timeline called "step".
for step in range(0, 64):
    rr.set_time_sequence("step", step)
    rr.log("scalar", rr.Scalar(math.sin(step / 10.0)))

File: docs/code-examples/all/scalar_simple.rs
//! Log a scalar over time.

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = rerun::RecordingStreamBuilder::new("rerun_example_scalar").spawn()?;

    // Set up plot styling: Logged timeless since it never changes and affects all timelines.
    rec.log_timeless("scalar", &rerun::SeriesPoint::new())?;

    // Log the data on a timeline called "step".
    for step in 0..64 {
        rec.set_time_sequence("step", step);
        rec.log("scalar", &rerun::Scalar::new((step as f64 / 10.0).sin()))?;
    }

    Ok(())
}
